client/components/saved-result.jsx

```diff
@@ -1,64 +1,211 @@
 import React from 'react';
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
 import AppContext from '../lib/app-context';
 
 export default class SavedResult extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { result: null };
+    this.state = {
+      result: null,
+      playlistName: ''
+    };
+
+    this.handleCreatePlaylist = this.handleCreatePlaylist.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleCancel = this.handleCancel.bind(this);
+    this.handleConfirm = this.handleConfirm.bind(this);
   }
 
   componentDidMount() {
+    const endpoint = this.props.libCategory.get('libCategory');
     const token = window.localStorage.getItem('user-jwt');
-    const endpoint = this.props.libCategory;
     const options = {
-      method: 'GET',
       headers: {
         'X-Access-Token': token
       }
     };
-    fetch('/api/user/library', options)
+    fetch(`/api/user/library/${endpoint}`, options)
       .then(res => res.json())
       .then(result => {
-        if (endpoint === 'songs') {
-          // eslint-disable-next-line array-callback-return
-          const savedList = result.map((item, index) => {
-            if (endpoint === 'songs') {
-              return (
-                <div key={item.id} data-id={index} className="row margin-neg">
-                  <div className="col-2 col-md-1 p-0">
-                    <div className="img-album my-1">
-                      <img
-                        src={item.albumCover}
-                        className="rounded img-fluid"
-                        alt={item.title} />
-                    </div>
+        // eslint-disable-next-line array-callback-return
+        const savedList = result.map((item, index) => {
+          if (endpoint === 'songs') {
+            return (
+              <div key={item.id} data-id={index} className="row margin-neg">
+                <div className="col-2 col-md-1 p-0">
```

Comment:
Here as well, we want to do our mapping in render. It could be in its own method, but the key is to just store a data model in state and not React elements.